homeassistant/components/pooldose/entity.py

```diff
@@ -73,11 +75,16 @@ def available(self) -> bool:
             return False
         # Check if the entity type exists in coordinator data
         platform_data = self.coordinator.data.get(self.platform_name, {})
-        return self.entity_description.key in platform_data
+        return (
+            isinstance(platform_data, dict)
+            and self.entity_description.key in platform_data
+        )
 
-    def get_data(self) -> dict | None:
+    def get_data(self) -> dict[str, Any] | None:
         """Get data for this entity, only if available."""
         if not self.available:
             return None
         platform_data = self.coordinator.data.get(self.platform_name, {})
-        return platform_data.get(self.entity_description.key)
+        if isinstance(platform_data, dict):
+            return platform_data.get(self.entity_description.key)
+        return None
```

Comment:
same, I am not sure why we have these extra checks

Reply:
Same as previous comment, it feels over-conversative, but mypi needs it to be passed.